lib/exception_handling.rb

```diff
@@ -193,7 +193,9 @@ def log_error_rack(exception, env, _rack_filter)
     def log_error(exception_or_string, exception_context = '', controller = nil, treat_like_warning: false, **log_context, &data_callback)
       ex = make_exception(exception_or_string)
       timestamp = set_log_error_timestamp
-      exception_info = ExceptionInfo.new(ex, exception_context, timestamp, controller || current_controller, data_callback)
+      exception_info = ExceptionInfo.new(ex, exception_context, timestamp,
+                                         controller: controller || current_controller, data_callback: data_callback,
+                                         log_context: logger.current_context_for_thread.deep_merge(log_context))
 
       if stub_handler
         stub_handler.handle_stub_log_error(exception_info.data)
```

Comment:
Just below a `log_context` is passed up to `execute_custom_log_error_callback`. I wonder if we want the plain `log_context` that was passed in (this is what's coded now) or if we want the `logger.current_context_for_thread.deep_merge(log_context))` from above?

Reply:
We want the plain version bellow because that is sent down to the logger which does it's own merge with the current thread context so you would be duplicating work.

Reply:
Sounds good. I think I will change this code to use a local variable to hold the `logger.current_context_for_thread.deep_merge(log_context)` just to help clarify that it's different (despite having the same keyword arg name).

Reply:
@jebentier Even better, I pushed the deep_merge responsibility down so that `log_context:` is always the same unmerged hash.
https://github.com/Invoca/exception_handling/pull/72/commits/e64c6c13a0a3692acbac909f79d3d119246ab91a